pkgs/applications/audio/jamesdsp/default.nix

```
@@ -2,18 +2,29 @@
 , mkDerivation
 , fetchFromGitHub
 , pipewire
+, pulseaudio
+, gst_all_1
 , glibmm
 , qmake
+, qtbase
+, qtsvg
+, wrapQtAppsHook
 , makeDesktopItem
 , pkg-config
 , libarchive
 , fetchpatch
+, copyDesktopItems
+, wrapGAppsHook
+, usePipewire ? false
+, usePulseaudio ? false
 }:
 
-mkDerivation rec{
+assert lib.asserts.assertMsg ((usePipewire||usePulseaudio)&&!(usePipewire&&usePulseaudio)) "You need to enable one and only one of pulseaudio or pipewire support";
```

This can be simplyfied to

```suggestion
assert lib.asserts.assertMsg (usePipewire && !usePulseaudio) "You need to enable one and only one of pulseaudio or pipewire support";
```

---

I think it should be `(usePipewire != usePulseaudio) `

---

Thats the same just written differently.

---

> if `usePipewire = false` and `usePulseaudio  = true`, (usePipewire && !usePulseaudio) -> (false && !true) -> false, but it should be true

---

You might be right.